thentos-tests/src/Thentos/Test/Core.hs

```diff
@@ -139,15 +139,15 @@ teardownBare (TS tcfg) = do
 
 -- * DBTS
 
-setupDB :: IO DBTS
+setupDB :: forall db . (db `Extends` DB, IsAcidic db, Eq (ThentosError db)) => IO (DBTS db)
```

Comment:
There is an explanation at `https://github.com/liqd/thentos/blob/master/docs/concepts/AcidPoly.lhs`